app/models/permission.rb

```diff
@@ -12,6 +12,10 @@ class Permission < ApplicationRecord
 
   def self.resources
     @all_resources ||= Permission.distinct.order(:resource_type).pluck(:resource_type).compact
+  rescue
+    return [] if ActiveRecord::Base.connection.migration_context.needs_migration?
+
+    raise
```

Comment:
This is needed due to https://github.com/theforeman/foreman/blob/a64edc9d661230e74b58ad98a857d93a3068f5b3/app/controllers/api/v2/template_inputs_controller.rb#L40

However, it doesn't appear to work:
Current output:
```console
# foreman-rake apipie:cache:index
2021-04-07 13:17:34 +0000 | Started
rake aborted!
Validator for #<Permission::ActiveRecord_Relation:0x000055db0727ac80> not found.
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:94:in `initialize'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `new'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `from_dsl_data'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:336:in `block in params_ordered'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:333:in `map'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:333:in `params_ordered'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:389:in `prepare_hash_params'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:329:in `initialize'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:315:in `new'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:315:in `build'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:34:in `block in find'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:33:in `each'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:33:in `find'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:93:in `initialize'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `new'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `from_dsl_data'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/method_description.rb:55:in `block in initialize'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/method_description.rb:54:in `map'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/method_description.rb:54:in `initialize'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:91:in `new'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:91:in `block in define_method_description'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:83:in `each'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:83:in `define_method_description'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/apipie_module.rb:34:in `method_missing'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/dsl_definition.rb:543:in `method_added'
/usr/share/foreman/app/controllers/api/v2/template_inputs_controller.rb:48:in `<class:TemplateInputsController>'
/usr/share/foreman/app/controllers/api/v2/template_inputs_controller.rb:3:in `<module:V2>'
/usr/share/foreman/app/controllers/api/v2/template_inputs_controller.rb:2:in `<module:Api>'
/usr/share/foreman/app/controllers/api/v2/template_inputs_controller.rb:1:in `<top (required)>'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:324:in `block in require'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:291:in `load_dependency'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:324:in `require'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:411:in `block in require_or_load'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:40:in `block in load_interlock'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies/interlock.rb:14:in `block in loading'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/concurrency/share_lock.rb:151:in `exclusive'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies/interlock.rb:13:in `loading'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:40:in `load_interlock'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:389:in `require_or_load'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:367:in `depend_on'
/usr/share/gems/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:280:in `require_dependency'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:440:in `load_controller_from_file'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:330:in `block in reload_documentation'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:329:in `each'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/application.rb:329:in `reload_documentation'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/apipie/apipie_module.rb:34:in `method_missing'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/tasks/apipie.rake:285:in `with_loaded_documentation'
/usr/share/gems/gems/apipie-rails-0.5.17/lib/tasks/apipie.rake:129:in `block (2 levels) in <top (required)>'
/usr/share/foreman/lib/tasks/apipie.rake:6:in `block (2 levels) in <top (required)>'
/usr/share/gems/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => apipie:cache
(See full trace by running task with --trace)
```

Looks like it doesn't immediately raise an exception but rather returns a lazy queryset. However, making it check before running can create a performance overhead. Not sure how to solve it.